### ISSUE TITLE
Web player: no avatars, names in transcript if a contributor is not visible

### ISSUE THREAD

**Issue by tomtjes** [podlove/podlove-ui#665]
### Expected behavior
Contributors' names and avatars should be displayed in the transcript tab, if marked as visible in contributors settings.

### Actual behavior
If an episode has a contributor marked as not visible, the names and avatars of visible contributors aren't in the transcript displayed either. This even occurs if the invisible contributor is not associated with a voice, hence doesn't appear in the transcript.

### System information (see `Podlove > Support` menu)
WordPress Version          5.5.1
Publisher Version          3.0.4
Web Player Version         player_v5

**Comment by tomtjes** [comment]
No, the invisible contributor does not have a voice part. All voice parts are assigned to visible contributors.

To provide more context in case that's helpful:
Visible contributors are in groups A and B with roles 1 and 2.
The invisible contributor is in group A with no role assigned in the episode.
When I remove the invisible contributor from the episode, the transcript appears with names and avatars as expected.

**Comment by tomtjes** [comment]
Hmmm, this is strange. I added the invisible contributor back to all our episodes, and it did not seem to have a negative effect, except on the latest episode https://recordofchange.com/2020/09/27/still-standing-home/ It now shows no contributors in the transcript anymore.

**Comment by sventropy** [comment]
@tomtjes I was able to reproduce the behavior with the data from your link. Thanks for sharing.

The reason, why no speakers are visible in the transcript is the following: Usually contributors are downloaded as an array of objects in the respective json document, example:

```json
{
      "id": "1",
      "name": "Tim Pritlove",
      "avatar": "https:\/\/forschergeist.de\/wp-content\/cache\/podlove\/47\/08928e3c26dcb1141d67ad75869619\/tim-pritlove_150x150.jpg",
      "role": {
        "id": "1",
        "slug": "moderation",
        "title": "Moderation"
      },
      "group": {
        "id": "1",
        "slug": "onair",
        "title": "On Air"
      },
      "comment": null
    },
    {
      "id": "68",
      "name": "Robert Ranisch",
      "avatar": "https:\/\/forschergeist.de\/wp-content\/cache\/podlove\/53\/bea5b7ea6e5c60379b268d8c2f7c31\/robert-ranisch_150x150.jpg",
      "role": {
        "id": "2",
        "slug": "guest",
        "title": "Gast"
      },
      "group": {
        "id": "1",
        "slug": "onair",
        "title": "On Air"
      },
      "comment": null
    },
...
```

In your case, for some reason the publisher sends an object with numeric properties to the player (excerpt): 

```json
"contributors": {
    "1": {
      "id": "13",
      "name": "Mike van Graan",
      "avatar": "https://recordofchange.com/wp-content/cache/podlove/e3/29849b43dfe9afa8e8744ea59b37d3/mike-van-graan_150x150.jpg",
      "role": null,
      "group": { "id": "2", "slug": "interviewees", "title": "Interviewees" },
      "comment": null
    },
    "2": {
      "id": "1",
      "name": "Matthias Jochmann",
      "avatar": "https://recordofchange.com/wp-content/cache/podlove/4f/8848f62994998010ddaa88f3dfb303/matthias-jochmann_150x150.jpg",
      "role": { "id": "5", "slug": "P", "title": "Producer" },
      "group": { "id": "1", "slug": "team", "title": "Team" },
      "comment": null
    },
    "3": {
      "id": "1",
      "name": "Matthias Jochmann",
      "avatar": "https://recordofchange.com/wp-content/cache/podlove/4f/8848f62994998010ddaa88f3dfb303/matthias-jochmann_150x150.jpg",
      "role": { "id": "1", "slug": "Host", "title": "Host" },
      "group": { "id": "1", "slug": "team", "title": "Team" },
      "comment": null
    },
   ...
  },
```

In that format, the [implementation of the transcript reducer](https://github.com/podlove/podlove-ui/blob/development/packages/player/sagas/transcripts.js#L89) does not find any speaker using [ramda find](https://ramdajs.com/docs/#find) as it can only cope with arrays.

That is the point, where I need a hint, probably from @eteubert: Is the publisher expected to send contributors in both formats (array and object) or is it a bug in the publisher?

I would be happy to keep digging based on your reply.

Cheers and happy weekend, Sven

**Comment by alexander-heimbuch** [comment]
@sventropy Thanks a lot for your debugging efforts 👍
